Скриншоты лб1

orocko-v2 opened the pull request.

orocko-v2 committed:
Скриншоты лб1

orocko-v2 committed:
Add files via upload

orocko-v2 committed:
Скриншоты лб3

orocko-v2 committed:
Скриншоты лб4

orocko-v2 commented:
ЛБ1-4

olglukhikh commented:
Здравствуйте, назовите ваши фамилию и имя.

orocko-v2 committed:
Скриншоты лб5

orocko-v2 commented:
Здравствуйте, Немеровченко Дмитрий, 8ПМ32.
Также прошу дополнительные задания для набора баллов

olglukhikh commented:
Лабораторная 1: 0б (принято; выполнено сильно после 2го дедлайна)
Лабораторная 2: 1,25б. Лучше сделать только номер лицевого счета первичным ключом, иначе нужно передавать все части первичного ключа как внешний ключ. (выполнено на 3ю неделю после 2го дедлайна)
Лабораторная 3: 2,5б (выполнено на 2ю неделю  после 2го дедлайна)
Лабораторная 4: 5б (выполнено на 1ю неделю  после 2го дедлайна)
Лабораторная 5: 5б (выполнено на 1ю неделю  после 2го дедлайна) Нет таблицы, где среди прочих атрибутов есть json, но зачту

olglukhikh commented:
Сделайте нормализованную (3НФ) ER-модель для 12 варианта 1 лабораторной, создайте таблицы и заполните базу по заданию 5й лабораторной (таблицы для нее не меняйте, создайте новые-копии). Причем выполните создание и заполнение таблиц запросами, идентификаторы в таблицах - типа uuid, заполнять с помощью функций. Сделайте минимум 2 запроса с соединениями в получившейся базе (хорошо, если будут участвовать атрибуты типа json). Компоненты блюда можно сделать одним текстовым атрибутом, но его нужно зашифровать.
Выполните запросы для 5, 6 и 7 вариантов 3й лабораторной.
По выполнению всех запросов выведите статистику запросов SELECT.
Сдавать можно по частям; внимательно читайте задания.

olglukhikh commented:
> Здравствуйте, Немеровченко Дмитрий, 8ПМ32. Также прошу дополнительные задания для набора баллов

Сделайте нормализованную (3НФ) ER-модель для 12 варианта 1 лабораторной, создайте таблицы и заполните базу по заданию 5й лабораторной (таблицы для нее не меняйте, создайте новые-копии). Причем выполните создание и заполнение таблиц запросами, идентификаторы в таблицах - типа uuid, заполнять с помощью функций. Сделайте минимум 2 запроса с соединениями в получившейся базе (хорошо, если будут участвовать атрибуты типа json). Компоненты блюда можно сделать одним текстовым атрибутом, но его нужно зашифровать.
Выполните запросы для 5, 6 и 7 вариантов 3й лабораторной.
По выполнению всех запросов выведите статистику запросов SELECT.
Сдавать можно по частям; внимательно читайте задания.

orocko-v2 committed:
ЛБ6 Скриншоты

orocko-v2 commented:
По поводу ЛБ2: если сделать первичным ключом только номер лицевого счета, могут возникнуть следующие проблемы:
1) В таблице могут встретиться 2 записи с разными лицевыми счетами, но одинаковыми адресами - соответственно, у одной квартиры 2 лицевых счета (что, вроде как, не может быть).
2) Площадь зависит от адреса квартиры, так как привязана к конкретной квартире с конкретным адресом. Соответственно, может возникнуть ситуация, при которой будут существовать 2 записи с одинаковыми адресами, но разными значениями площадей.
Как вариант, можно все поля адреса вынести в отдельную таблицу и связать при помощи адрес_айди, при этом в таблице Квартира данный адрес сделать уникальным. Тогда одной квартире будет соответствовать один номер лицевого счета, и можно сказать, что площадь зависит от лицевого счета -> сделать лицевой счет единственным первичным ключом. (только я поздно об этом догадался)

orocko-v2 committed:
Нормализованная ER-модель (12 вариант)

orocko-v2 committed:
Запросы для 3 ЛР (5,6,7 вар)

Запрос 5 вар 6 совпадает с запросом 1 вар 7, соответственно, загрузил один скриншот на оба запроса

orocko-v2 committed:
Статистика запросов select

orocko-v2 committed:
Delete доп1.png

orocko-v2 committed:
(Исправлено) ER-модель (12 вариант)

orocko-v2 committed:
Создание и заполнение таблиц

Выполнено создание и заполнение таблиц, идентификаторы - типы uuid, 2 запроса с соединениями (join), один из запросов с jsonb атрибутом, который был зашифрован

olglukhikh commented:
Лабораторная 6: 10 б
Доп задание принято, баллы к лабораторным доставлю.
Рекомендация: читайте внимательно задания. Был пропущен минимум один запрос, зашифрована не та информация (это не повлияет на количество баллов).
По 2 лаб.: номер лицевого счета может повторяться в одной квартире, если квартиросъемщиков несколько (почему-то решили не скидываться, а сделать два договора). Этот комментарий был к тому, что для foreign key первичный ключ должен передаваться в полном объеме, то есть в вашем случае - составной ключ из номера и адреса.
